prSHARK/backends/github.py

```diff
@@ -202,8 +209,42 @@ def _fetch_all_pages(self, base_url):
 
     def run(self):
         """Executes the complete workflow, fetches all data and saves it into the MongoDB."""
+        last_system = PullRequestSystem.objects.filter(project_id=self._p.id).order_by('-collection_date').first()
+        self.last_system_id = last_system.id if last_system else None
+
+        self.pr_system = PullRequestSystem(project_id=self._p.id, url=self.config.tracking_url, collection_date=datetime.datetime.now())
+        self.pr_system.save()
+
         self.parse_pr_list(self.fetch_pr_list())
 
+        self.save_prs()
+
+    def save_prs(self):
+        """
+         Save pull requests and related data to the database.
+
+        """
+        self._log.info('Saving pull requests')
+        for pr_id, pr in self.parsed_prs['prs'].items():
+            if pr_id in self.pr_diff and self.pr_diff[pr_id]:
```

Comment:
You can probably avoid a deep nesting if you turn the condition around, e.g., `if pr_id not in self.pr_diff: continue`. Could pr_id be in self.pr_diff with it being None or False?